pages/k8s/logging.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "Logging | Canonical Distribution of Kubernetes&reg;"
+keywords: juju, logging, debug-log
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: logging.html
+layout: base
+toc: False
+summary: This page is currently a work in progress. For existing documentation, please visit <a href="https://kubernetes.io/docs/getting-started-guides/ubuntu/"> https://kubernetes.io/docs/getting-started-guides/ubuntu/ </a>
+---
+
+# Logging
+
+<div class="p-notification--information">
+  <p class="p-notification__response">
+    <span class="p-notification__status">Note:</span>
+This documentation assumes you are using version 2.4.0 or later of
+<strong>Juju</strong> . If you are using an earlier version you should
+check the  <a href="https://docs.jujucharms.com/stable/en/troubleshooting-logs">
+ relevant <emphasis>Juju</emphasis> documentation </a> as some of the associated
+ commands have changed.
+  </p>
+</div>
+
+
+Your **CDK** deployment has centralised logging set up as default. Each unit in
+your cluster automatically sends logging information to the controller based on
+the current logging level.  You can use the **Juju** command line to easily
+inspect these logs and to change the logging level, as explained below.
+
+## Viewing logs
+
+To view the logs from the current controller and model, simply run:
+
+```
+juju debug-log
+```
+
+The default behaviour is to show the last 10 entries and to tail the log (so
+you will need to terminate the command with `Ctrl-C`).
+
+The output is in the form:
+
+`<entity> <timestamp> <log-level> <module>[:<line-no>] <message>`
+
+For example, a typical line of output might read:
+
+```
+unit-kubernetes-master-0: 18:04:11 INFO juju.cmd running jujud [2.4.2 gc go1.10]
+```
+
+The entity is the unit, machine or application the message originates from (in
+this case _kubernetes-master/0_). It can be very useful to filter the output
+based on the entity or log level, and the `debug-log` command has many options.
+For a full description, run the command `juju help debug-log` or see the
+[**Juju** documentation][juju-logging]. Some useful examples are outlined below.
+
+### Useful examples
+
+View the last 100 entries and tail the log:
+
+```bash
+juju debug-log -n 100
+```
+
+Show the last 20 entries and exit:
+
+```bash
+juju debug-log -n 20 --no-tail
+```
+
+Replay the log from the very beginning, but filter to logs from kubernetes-worker/0:
+
+```bash
+juju debug-log --replay --include=kubernetes-worker/0
+```
+
+
+## Viewing logs on a machine
```

Comment:
Should we mention logging of Kubernetes itself? Maybe even just a link to the k8s docs. Usually if juju debug-log doesn't help, it is time to look at the api server and kubelet logs.

Reply:
I agree with the thought here. This doc covers logging of the infrastructure, but not logging of the k8s cluster components (syslog from the k8s units), or logging of the container workloads running on the cluster. The blog post by 
@kwmonroe covers those, so we could pull those in. Doesn't have to be in this PR, but we'll need it.